yt_dlp/extractor/nuum.py

```diff
@@ -0,0 +1,133 @@
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    int_or_none,
+    parse_iso8601,
+    traverse_obj,
+    try_get,
+)
+
+
+class NuumBaseIE(InfoExtractor):
+
+    def _fetch(self, path, video_id, description, query={}):
+        response = self._download_json(
+            f'https://nuum.ru/api/v2/{path}', video_id, query=query,
+            note=f'Downloading {description} metadata',
+            errnote=f'Unable to download {description} metadata')
+        error = response.get('error')
+        if error:
+            raise ExtractorError(f'{self.IE_NAME} returned error: {error}', expected=True)
+        return response.get('result')
+
+    def _extract_thumbnails(self, thumbnails_dict):
+        return [{
+            'url': url,
+            'preference': index,
+        } for index, url in enumerate(
+            traverse_obj(thumbnails_dict, (('small', 'medium', 'large'),))) if url]
+
+    def _get_container(self, url):
+        container_id = self._match_id(url)
+        return self._fetch(
+            f'media-containers/{container_id}', container_id, 'media container')
+
+    def _get_media_url(self, media_meta):
+        media_archive_url = media_meta.get('media_archive_url')
+        if media_archive_url:
+            return media_archive_url, False
+        return media_meta['media_url'], True
+
+    def _real_extract(self, url):
+        container = self._get_container(url)
+        stream = traverse_obj(container, ('media_container_streams', 0))
+        media = try_get(stream, lambda x: x['stream_media'][0])
+        if not media:
+            raise ExtractorError('Cannot extract media data', expected=True)
+        media_meta = media.get('media_meta')
+        media_url, is_live = self._get_media_url(media_meta)
+        video_id = media.get('media_id') or container.get('media_container_id')
+        formats, subtitles = self._extract_m3u8_formats_and_subtitles(media_url, video_id, 'mp4')
+        return {
+            'id': str(video_id),
+            'title': container.get('media_container_name') or self._og_search_title(self._download_webpage(url, video_id)),
+            'description': container.get('media_container_description'),
+            'thumbnails': self._extract_thumbnails(media_meta.get('media_preview_images' if is_live else 'media_preview_archive_images')),
+            'timestamp': parse_iso8601(container.get('created_at')),
+            'view_count': int_or_none(stream.get('stream_current_viewers' if is_live else 'stream_total_viewers')),
+            'is_live': is_live,
+            'formats': formats,
+            'subtitles': subtitles,
+        }
+
+
+class NuumLiveIE(NuumBaseIE):
+    IE_NAME = 'nuum:live'
+    _VALID_URL = r'https?://nuum\.ru/channel/(?P<id>[^/#?]+)'
```

Comment:
I think attempting to download the livestreams here is unexpected.
In my opinion, it would make more sense to handle this like the YouTube extractor:
`https://nuum.ru/channel/channel_name` - return all the tabs (the links below)
`https://nuum.ru/channel/channel_name/streams` - return all currently ongoing streams (not sure if there can be multiple at once)
`https://nuum.ru/channel/channel_name/videos` - return all videos
`https://nuum.ru/channel/channel_name/clips` - return all clips

You don't need to implement all that if you don't want to.

Reply:
To be able to download livestreams it requires a direct link to it (for example: `https://nuum.ru/streams/1592713-7-days-to-die`).
So it's not possible to download a livestream whenever it becomes live without knowing a link to it.
That's why I made it like this.

Is it currently possible on youtube?

Reply:
With a YouTube channel, `https://www.youtube.com/@channel_name/live` would lead to the current livestream.
Since `https://nuum.ru/channel/channel_name/streams` shows all streams, it makes more sense to use that to return all currently ongoing streams (even if it's just one). That way, you also have a constant URL to fetch any potential livestreams.

Reply:
`/streams` - ongoing streams (if any) and all VODs.
`/videos` - videos uploaded by users.

VODs are available from the same link after a stream ends.
I think they are available only for some users and stored for 14 days.

So if I provide `/streams` link it will download ongoing stream and then continue downloading VODs, which I don't expect if I want to download only live streams.

Reply:
Okay yeah, I initially didn't see any channels with archived livestreams. I think your initial idea is fine, then.
`https://nuum.ru/channel/dankon_` - url result for current livestream or `UserNotLive` error.
`https://nuum.ru/channel/dankon_/clips` - playlist for all clips
`https://nuum.ru/channel/dankon_/videos` - playlist for all videos
`https://nuum.ru/channel/dankon_/streams` - playlist for all stream recordings

You don't need to implement the playlists.